sunscreen_runtime/debugger-frontend/src/App.tsx

```diff
@@ -200,25 +276,69 @@ const App = () => {
           { source: 1, target: 2, directed: true, arrowhead: 'normal' }
         ]
       }
-      setGraph(lineNumber !== 1 ? graph : exGraph)
-    }, [setLine, setGraph]
+      setGraph(lineNumber !== 1 ? graph : dataToGraph(await fetch(`/sessions/${session}`).then(d => d.json())))
+    }, [setLine, setGraph, session]
   )
 
   const updateSelection = useCallback(
-    (selection, e) => {select(selection); console.log(selection.nodes?.values().next().value)}, [select]
+    async (selection, e) => {
+      select(selection); 
+      const node = selection.nodes?.values().next().value;
+      console.log(node)
+      if (node != null) {
+        if (session.split('_')[0] == "fhe") {
+          setInfo({
+            ...selection.nodes?.values().next().value, 
+            ...(await fetch(`sessions/${session}/${node.id}`).then(d => d.json())).Bfv,
+            stacktrace: filterStackTrace(await fetch(`sessions/${session}/stacktrace/${node.id}`).then(d => d.json()))
+          })
+        } else {
+          setInfo({
+            ...selection.nodes?.values().next().value, 
+            value: (await fetch(`sessions/${session}/${node.id}`).then(d => d.json())).Zkp,
+            stacktrace: filterStackTrace(await fetch(`sessions/${session}/stacktrace/${node.id}`).then(d => d.json()))
+          })
+        }
+        
+      } else {
+        setInfo({id: "no node selected"})
+      }
+    }, [select, session]
+  )
+
+  const updateSession = useCallback(
+    (event) => {
+      const newSession = event.target.value
+      
+      setSession(newSession)
+    }, [setSession]
```

Comment:
`setSession` doesn't do anything as a dependency, and this should not be a `useCallback`. To be clear, it can just be

```js
const updateSession = (event) => { setSession(event.target.value) }
```

it could even just be inlined below.